### PR TITLE
architecture

### PR DESCRIPTION
**Pull request by tristancabel**
- dummy_archi2 and spdMatrices inheritance

### CODE REVIEW

File: examples/dummy_archi2.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        if len(metrics) > 1:
+            res = {}
+            for metric in super().getMetrics():
+                res[metric.name()] = metric() # TODO tangent_vec
+            return res
+        elif len(metrics) == 1:
+            return metrics[0]()
+        else :
+            print("no metric")
+            return 0
```
Comment (nguigs) [inline examples/dummy_archi2.py:107]: Would these conditions have to be added to every method of the `Metric` class?

File: examples/dummy_archi2.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    def __init__(self, metrics : List[Metric]):
+        self.metrics = metrics
+
+    def getMetrics(self) -> List[Metric]:
+        return self.metrics
+
+    @abstractmethod
+    def exp(self, tangent_vec) -> Union[int, Dict[str, int]]:  # TODO return also only str 
+        pass
```
[the remark this answers is not in the document]
Comment (tristancabel) [reply examples/dummy_archi2.py]: done in last commit

File: examples/dummy_archi2.py (hunk 1 above)
Comment (tristancabel) [reply examples/dummy_archi2.py]: solved

**Comment by tristancabel** [comment]
@nguigs , what do you think?

**Comment by nguigs** [comment]
Hi @tristancabel, thanks for the notebook! It is nice from the developer's point of view. There is also the user point of view, e.g. I want to compare two metrics on some manifold, how do I do? Maybe you could re use this [example on se2](https://github.com/geomstats/geomstats/blob/master/examples/plot_geodesics_se2.py) and see how it adapts to the new architecture?

**Comment by tristancabel** [comment]
> Hi @tristancabel, thanks for the notebook! It is nice from the developer's point of view. There is also the user point of view, e.g. I want to compare two metrics on some manifold, how do I do? Maybe you could re use this [example on se2](https://github.com/geomstats/geomstats/blob/master/examples/plot_geodesics_se2.py) and see how it adapts to the new architecture?

done [here](https://github.com/geomstats/geomstats/pull/937/files#diff-ca7230c96e75ce84def5bcda4c18c62b95a2151b8b5712941769a4c7463e2ffbR36)

**Comment by ninamiolane** [comment]
Closing this stalled PR: @nguigs has refactor the architecture of the `geometry` module.

**Pull request closed without merging**